File: 프로그래머스/2/42842. 카펫/카펫.cpp
#include <vector>
#include <cmath>

using namespace std;

vector<int> solution(int brown, int yellow) {

    int total = brown + yellow;
    vector<int> result;
    // 최소 세로길이인 3부터 시작
    for(int width = 3 ; width <= sqrt(total) ; width++) {
        if(total % width == 0) {
            int height = total / width;
            int curBrown = (width + height) * 2 - 4;
            int curYellow = total - curBrown;
            
            if(curBrown == brown && curYellow == yellow) {
                result.push_back(height);
                result.push_back(width);
                break;
            }
        }
    }
    
    return result;
}

File: 프로그래머스/2/42842. 카펫/카펫.java
class Solution {
    
    int weight = -1;
    int height = -1;
    
    
    public int[] solution(int brown, int yellow) {
        
        int wPh = (brown + 4) / 2; // 가로 + 세로
        
        for(int h = 1 ; h <= wPh / 2 ; h++) {
            int w = wPh - h;
            if(h * w - brown == yellow){
                weight = w;
                height = h;
                break;
            }
        }
        
        return new int[]{weight, height};
    }
}